src/main/java/emissary/util/io/GreedyResourceReader.java

```diff
@@ -0,0 +1,134 @@
+package emissary.util.io;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.StringUtils.substringBeforeLast;
+
+/**
+ * A {@link ResourceReader} extended to find data files based purely on their location, ignoring the default naming
+ * convention used by the {@link ResourceReader#findDataResourcesFor(Class)} method.
+ * <p>
+ * This class is primarily used to find payload files for Identification tests that can benefit from more
+ * content-representative file names.
+ * </p>
+ */
+public class GreedyResourceReader extends ResourceReader {
```

Comment:
Should this class exist in the test package?

Reply:
You could argue either way, but I like it where it is.  As a class that supports a test (as opposed to containing tests), it seems reasonable to be in main/util/io.